src/format/Format.cc

```diff
@@ -867,7 +871,7 @@ Format::Format::assemble(MemBuf &mb, const AccessLogEntry::Pointer &al, int logS
             break;
 
         case LFT_ICAP_TR_RESPONSE_TIME:
-            outtv = al->icap.trTime;
+            outtv = al->icap.trTime(recordTime);
```

Comment:
AFAICT ICAP ModXact ALE object and the corresponding master/HTTP transaction ALE object are the same. ICAP code uses AccessLogEntry::IcapLogEntry section of that object (in some cases), but the object is the same in both contexts.

Let's consider an ordinary HTTP master transaction that went through an ordinary ICAP REQMOD transaction shortly after Squid received an HTTP request, then went to the origin server, responded to the client, and is now logging `%icap::tr` to access_log.

Before this change, logged `%icap::tr` reflected REQMOD transaction response time.

After this change, logged `%icap::tr` reflects time passed since that (already completed!) REQMOD transaction started. This logged value may be significantly larger than the actual REQMOD response time we previously logged.

N.B. None of the above is meant to imply that official code is problem-free in this area. However, what I am describing above feels like a new problem that this PR is adding.

Unless I am missing something, we have to distinguish three ICAP trTime states:

* ICAP transaction has not started -- %icap::tr should expand to `-`.
* ICAP transaction has started but not finished -- %icap::tr grows with passage of time.
* ICAP transaction has started and finished -- %icap::tr should not change with further passage of time.

We should also be careful with _multiple_ ICAP transactions per master transaction. I think it is OK for ICAP trTime to reflect the last _completed_ ICAP transaction even if there was the next ICAP transaction started for the same master transaction, especially if that is what official code does now.

Reply:
> ICAP ModXact ALE object and the corresponding master/HTTP transaction ALE object are the same.

I'm not sure what is "ICAP ModXact ALE object" here. ModXact's constructor gets a master ALE as an argument and stores it into ModXact::alMaster, so alMaster and HTTP transaction ALE are the same. That alMaster is not used for logging, though.

>  imply that official code is problem-free in this area

There are probably problems there: the official code cannot log %icap::tr to access.log and %http::tr to icap.log because HTTP's start_time is not set in the ICAP's ALE and ICAP's start_time is not set in the HTTP's ALE (in both cases '0' is logged).

> we have to distinguish three ICAP trTime states

Ok, I'm adding IcapLogEntry::stop_time to distinguish these cases.

Reply:
> ICAP's start_time is not set in the HTTP's ALE

Moreover, I used debugger to check whether other IcapLogEntry fields are filled for the HTTP's ALE. I found only one such field - icap.processingTime. It is set in prepareLogWithRequestDetails() (which is called from logRequest() and ModXact::finalizeLogInfo()) and %icap::tt works for both access.log and icap.log (showing the same value).